bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2100,15 +2173,12 @@ public int hashCode() {
 
 	@Override
 	public boolean equals(Object otherProvider) {
-		return otherProvider instanceof AbstractImageProviderWrapper aip //
-				&& getProvider().equals(aip.getProvider());
-	}
-
-	protected void destroy() {
+		return otherProvider instanceof DynamicImageProviderWrapper aip //
+				&& getProvider() != null && getProvider().equals(aip.getProvider());
```

Comment:
Since you also seem to allow `null` values, shouldn't you consider the case were both providers are `null`?

```suggestion
		return otherProvider instanceof DynamicImageProviderWrapper aip //
				&& Objects.equals(getProvider(), aip.getProvider());
```

Reply:
Adressed in https://github.com/eclipse-platform/eclipse.platform.swt/commit/dd9b823703a175c28d4e4a5efc725cb7f20cbb2b

Additional remark:  I had a look and getProvider actually can not return null. That is implicitly prevented be the checkProvider call in the constructor